share/picongpu/tests/CurrentDeposition/lib/python/test/CurrentDeposition/grid_class.py

```diff
@@ -0,0 +1,207 @@
+"""
+This file is part of the PIConGPU.
+
+Copyright 2023 PIConGPU contributors
+Authors: Hannes Wolf
+License: GPLv3+
+
+Intoduces a grid class. A particle with defined assignment function can do a
+move (also randomly) on the grid from one position the another one. The
+current deposition and current density vector fields can then be calculated.
+"""
+
+
+import numpy as np
+from AfuncW import NGP, CIC, TSC, PSQ, W
+import random
+
+
+class grid:
+    def __init__(self, order):
+        self.order = order
+        if order == 0:
+            self.func = NGP
+        elif order == 1:
+            self.func = CIC
+        elif order == 2:
+            self.func = TSC
+        elif order == 3:
+            self.func = PSQ
+        else:
+            print("There are just assignment functions up to order 3")
+            print("implemented yet /n")
+
+    def create_grid(self):
+        """creates grid appropiate to the order of the assignement function
+
+        returns meshgrids for each direction
+        """
+        order = self.order
+        # number of cells, that a-func fits in
+        numCells = order + 3 + 1  # +1 for recursive current calculation [i-1]
+        # creates the grids axis
+        axis = np.arange(0, numCells, 1, dtype=float)
+        # creates grid, meshgrid for each direction
+        grid_x, grid_y, grid_z = np. meshgrid(axis, axis, axis)
+
+        return grid_x, grid_y, grid_z
+
+    def ran_particle_step(self):
+        """Randomly selects start and end point of a macro particle.
+        Randomly selects whether the endpoint is located in the initial cell
+        or not;
+        """
+        order = self.order
+        num_cells = order + 3 + 1
+
+        # choose Startcell in the middle of the grid
+        start_x_zelle = int((num_cells-1) / 2)
+        start_y_zelle = int((num_cells-1) / 2)
+        start_z_zelle = int((num_cells-1) / 2)
+
+        # random

```

Comment:
please explain in a bit more detail what this random contribution does

Reply:
this part is just for testing. I remove it.